### ISSUE TITLE
December 15 build not working on firefox 44 beta

### ISSUE THREAD

**Issue by rayman89** [piroor/suspendtab#43]
Not working at all. There are no tab menu items. It does not auto unload tabs or anything.

**Comment by linvinus** [comment]
please try this patch https://github.com/linvinus/suspendtab/commit/7191b36b9042b026363f511c5b2d7d30d756ec98

**Comment by rayman89** [comment]
Is this patch based on nightly version of suspendtab tab or release? I tested it on nightly and tab menus now are there but they don't do anything..

**Comment by linvinus** [comment]
path based on git master version, you may clone my forked version and then build as usual.
if suspendtab appear in menu then it should working.
try to open several tabs, then on current active tab, run "suspend other tabs", after that click on any other tab, if it was suspended then you will see page loading progress same as you just open it first time.

**Comment by rayman89** [comment]
Sorry I'm not a dev I don't know how to do that. Could you provide me with either the already done xpi or the "git master version" xpi?

I tested "suspend other tabs" on a clean profile and it didn't work. Auto suspend tabs doesn't work either. Wouldn't be the first add-on i tested to have a menu without functionality.

**Comment by linvinus** [comment]
there is ftp://linvinus.ru/forum/firefox/suspendtab_noupdate.xpi

**Comment by rayman89** [comment]
Tested that and it's the same. It does not work. The context menu on the tabs do show but they do nothing and the tabs do not automatically unload/suspend either. (it was tested on a new clean profile firefox 44 beta x64 on windows 7 x64)

**Comment by linvinus** [comment]
you are right! there is updated version  ftp://linvinus.ru/forum/firefox/suspendtab_noupdate.xpi
now it working, changes available there https://github.com/linvinus/suspendtab/commit/5f7d4008ffc1d591a39cd4fb38f4e04fd33b14fe

**Comment by rayman89** [comment]
Tested it and it's really buggy. Background pages don't load by default when that's an option nto default behavior and when I try to load a suspended page or this never loaded pages it wont load/render and if i select an other tab the tittle of the previous tab keeps switching between loading and the loaded title really fast but it never actually loads or renders the page.

**Comment by linvinus** [comment]
yep, thanks for testing, after while i also observe that.
hope to resolve that.
suspend is working, resume is buggy.

**Comment by linvinus** [comment]
please try new version version ftp://linvinus.ru/forum/firefox/suspendtab_noupdate.xpi
changes  https://github.com/linvinus/suspendtab/commit/35d36f170081fb6cf89b7e99505741f5d949af19
on my system working fine

**Comment by rayman89** [comment]
Has the same exact issue as the one before.(it was tested on a new clean profile firefox 44 beta x64 on windows 7 x64)

**Comment by linvinus** [comment]
finally now FF 44  working.
please try new version version ftp://linvinus.ru/forum/firefox/suspendtab_noupdate.xpi
changes https://github.com/linvinus/suspendtab/commit/56fa835c10ea0f745ac7e26c3c0d57cdba501a44

don't forget to disable automatically update in extension preferences!!! 
extensions->suspendtab->More->Automatic Updates=Off

**Comment by rayman89** [comment]
Works properly now. Thanks a lot.

**Comment by piroor** [comment]
I've feedback some changes from https://github.com/linvinus/suspendtab/commits/master

**Comment by linvinus** [comment]
what about this changes? https://github.com/linvinus/suspendtab/commit/35d36f170081fb6cf89b7e99505741f5d949af19

as i understand now FF will call SessionStoreInternal.restoreTabContent(aTab); by it self on tab select event if browser.__SS_restoreState = TAB_STATE_NEEDS_RESTORE;

**Comment by piroor** [comment]
Thanks! I missed that the point. 9e905e850f9531156d74e47a57bbedee7c1866ed is my version. For compatibility I think I should keep other codes to dispatch custom events.

**Comment by linvinus** [comment]
ok, it is up to you.

also it will be good to save tab icon, for suspended tab, so it will looks the same as unsuspended.

for example, currently suspended tabs looks differently, on the left suspended tabs without icon, on the right not suspended.
[![screenshot](http://storage4.static.itmages.ru/i/16/0128/s_1454005390_6994355_a590c49312.png)](http://itmages.ru/image/view/3612492/a590c493)

(changes on line 350 in suspendtab-internal.js)
```
    // Update the persistent tab state cache with |state| information.
    TabStateCache.update(browser, {
		  history: {entries: state.entries, index: state.index},
		  scroll: state.scroll || null,
		  storage: state.storage || null,
		  formdata: state.formdata || null,
		  disallow: state.disallow || null,
		  pageStyle: state.pageStyle || null

      // This information is only needed until the tab has finished restoring.
      // When that's done it will be removed from the cache and we always
      // collect it in TabState._collectBaseTabData().
      image: state.image || "",
      userTypedValue: state.userTypedValue || "",
      userTypedClear: state.userTypedClear || 0
    });

```

**Comment by piroor** [comment]
Thanks, merged at b20d44a6d404d547ed50e5fa93c0a149ea198e0f.

**Comment by linvinus** [comment]
this bug should be closed as resolved, if author don't agree with that then he could reopen it.

**Comment by piroor** [comment]
Now it's under doogfooding. I'll release a new version, if it works correctly on my daily environment.

**Comment by linvinus** [comment]
on my system also working fine
